sdk/azcore/runtime/policy_retry.go

```diff
@@ -59,13 +59,33 @@ func setDefaults(o *policy.RetryOptions) {
 }
 
 func calcDelay(o policy.RetryOptions, try int32) time.Duration { // try is >=1; never 0
-	delay := time.Duration((1<<try)-1) * o.RetryDelay
+	// avoid overflow when shifting left
+	factor := time.Duration(math.MaxInt64)
+	if try < 63 {
+		factor = time.Duration(int64(1)<<try - 1)
```

Comment:
If we don't have a linting rule about redundant parentheses, I'd do this just so people don't have to check the order of evaluation like I did 😁 
```suggestion
		factor = time.Duration(int64(1<<try) - 1)
```